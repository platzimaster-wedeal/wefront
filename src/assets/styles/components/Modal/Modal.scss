@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../partials/shadows.scss';
@import '../../Breakpoints.scss';

.modal {
  box-sizing: border-box;
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  width: 90%;
  height: 60vh;
  background-color: #fff;
  padding: $space-size-normal;
  border-radius: 15px;
  box-shadow: $shadow-large;
  overflow-y: scroll;

  @include media(tablet) {
    width: 54%;
    height: auto;
    max-height: 96vh;
  }
  &__title {
    display: flex;
    flex-flow: column;
    align-items: center;
    justify-content: center;
    color: $color-primary;
    width: 100%;
    border-bottom: 3px solid $color-primary;
    padding: $space-size-extra-small;
    margin: $space-size-normal 0;

    & h2 {
      @include font-size($font-size-normal);
      text-align: center;

      @include media(tablet) {
        @include font-size($font-size-medium);
      }
    }
  }

  &__action {
    position: absolute;
    top: 6%;
    right: 6%;
  }

 
}