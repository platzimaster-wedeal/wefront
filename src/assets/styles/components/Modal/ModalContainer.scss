@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../partials/shadows.scss';
@import '../../Breakpoints.scss';

.modal-container {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: center;
  position: fixed;
  background-color: $color-black-60;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  z-index: 999;
}