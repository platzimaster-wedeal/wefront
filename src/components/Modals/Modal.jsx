import React from "react";

// Styles
import "../../assets/styles/components/Modal/Modal.scss";

// Components
import Button from "../Buttons/Button";

const Modal = ({ children, title, onClose }) => {
 return (
  <div className="modal">
   <div className="modal__title">
    <h2>{title || "Title of Modal"}</h2>
   </div>
   <div className="modal__content">{children}</div>
   <div className="modal__action">
    <Button active onClick={onClose}>
     Close
    </Button>
   </div>
  </div>
 );
};

export default Modal;
