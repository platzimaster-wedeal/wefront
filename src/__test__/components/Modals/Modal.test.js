import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Modal from "../../../components/Modals/Modal";

describe("Test for the component Modal :D", () => {
 const ModalMounted = mount(<Modal />);
 const ModalCreated = create(<Modal />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(ModalCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(ModalMounted.length).toEqual(1);
  });
  test("Should the title of the component be a string", () => {
   expect(typeof ModalMounted.find(".modal__title h2").text()).toEqual(
    "string"
   );
  });
  test("Should the title of the component be a string", () => {
   expect(
    mount(<Modal title="Test Modal" />)
     .find(".modal__title h2")
     .text()
   ).toEqual("Test Modal");
  });
  test("Should the action of the component be a string", () => {
   expect(typeof ModalMounted.find(".modal__action button").text()).toEqual(
    "string"
   );
  });
  test("Should the action of the component be a string", () => {
   expect(ModalMounted.find(".modal__action button").text()).toEqual("Close");
  });
 });
});
