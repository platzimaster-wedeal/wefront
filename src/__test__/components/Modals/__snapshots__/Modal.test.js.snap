// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component Modal :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="modal"
>
  <div
    className="modal__title"
  >
    <h2>
      Title of Modal
    </h2>
  </div>
  <div
    className="modal__content"
  />
  <div
    className="modal__action"
  >
    <button
      className="button-actionCA undefined"
      style={
        Object {
          "backgroundColor": " hsl(152,59%,94%)",
          "color": " hsla(240,27%,12%,.6)",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
</div>
`;
